src/main.py

```diff
@@ -166,13 +148,14 @@ def on_button_ok_clicked(self, widget):
     def on_button_cancel_clicked(self, widget):
         self.destroy()
 
-    def set_source_state(self, state):
-        self.label30.set_sensitive(state)
-        self.combobox_source.set_sensitive(state)
+    def set_selection_mode(self, random: bool) -> None :
+        if random:
+            (self.treeview_source.get_selection()).set_mode(Gtk.SelectionMode.MULTIPLE)
+        else:
+            (self.treeview_source.get_selection()).set_mode(Gtk.SelectionMode.BROWSE)
```

Comment:
The selection mode of the TreeView is determined by the usage mode of the app: If random selection mode is chosen then the TreeView allows multiple selection. Otherwise, on one selection is allowed from the sources in the list.